src/jwtOAuth2.ts

```diff
@@ -0,0 +1,25 @@
+import OAuth2, { OAuth2Config } from './oauth2';
+
+export type JwtOAuth2Config = OAuth2Config & {
+  privateKey?: string;
+  privateKeyFile?: string;
+  authCode?: string;
+  refreshToken?: string;
+  loginUrl?: string;
+  username?: string;
+};
+
+export class JwtOAuth2 extends OAuth2 {
+  constructor(config: OAuth2Config) {
+    super(config);
+  }
+
+  public jwtAuthorize(innerToken: string): Promise<any> {
+    return super._postParams({
+      grant_type: 'urn:ietf:params:oauth:grant-type:jwt-bearer',
+      assertion: innerToken,
+    });
+  }
+}
+
+export default JwtOAuth2;
```

Comment:
nit: you can move this to the class definition to avoid this line.  I.e., line 12 above becomes:
`export default class JwtOAuth2 extends OAuth2 {`